streams/src/main/java/org/apache/kafka/streams/state/internals/KeyAndJoinSideSerializer.java

```diff
@@ -36,13 +37,14 @@
         this.keySerializer = keySerializer;
     }
 
+    @SuppressWarnings("unchecked")
     @Override
-    public void setIfUnset(final Serializer<K> defaultKeySerializer, final Serializer<Void> defaultValueSerializer) {
+    public void setIfUnset(final SerdeGetter getter) {
         if (keySerializer == null) {
-            keySerializer = Objects.requireNonNull(defaultKeySerializer, "defaultKeySerializer cannot be null");
+            keySerializer = (Serializer<K>) Objects.requireNonNull(getter.keySerde().serializer(), "defaultKeySerializer cannot be null");
```

Comment:
We can remove `requireNonNull` here, because `getter.keySerde()` would already throw a `ConfigException` if the default serde is null.